client/transaction-pool/src/api.rs

```diff
@@ -105,7 +105,7 @@ where
 				#[allow(deprecated)] // old validate_transaction
 				runtime_api.validate_transaction_before_version_2(&at, uxt)
 			};
-			let res = res.map_err(|e| Error::RuntimeApi(format!("{:?}", e)));
+			let res = res.map_err(|e| Error::RuntimeApi(format!("{}", e)));
```

Comment:
You need to change the where bound above. From:
```
sp_api::ApiErrorFor<Client, Block>: Send
```

to

```
sp_api::ApiErrorFor<Client, Block>: Send + std::fmt::Display
```

Reply:
Thanks!

Reply:
```suggestion
			let res = res.map_err(|e| Error::RuntimeApi(e.to_string()));
```